go/vt/vttablet/tabletserver/throttle/throttler.go

```diff
@@ -518,6 +520,43 @@ func (throttler *Throttler) Operate(ctx context.Context) {
 	}
 }
 
+func (throttler *Throttler) generateTabletHTTPProbeFunction(ctx context.Context, clusterName string, probe *mysql.Probe) (probeFunc func() *mysql.MySQLThrottleMetric) {
+	if probe.TabletHost == "" {
+		// nil function means no override; throttler will use default probe behavior, which is to open a direct
+		// connection to mysql and run a query
+		return nil
+	}
+	return func() *mysql.MySQLThrottleMetric {
+		// Hit a tablet's `check-self` via HTTP, and convert its CheckResult JSON output into a MySQLThrottleMetric
+		mySQLThrottleMetric := mysql.NewMySQLThrottleMetric()
+		mySQLThrottleMetric.ClusterName = clusterName
+		mySQLThrottleMetric.Key = probe.Key
+
+		tabletCheckSelfURL := fmt.Sprintf("http://%s:%d/throttler/check-self?app=vitess", probe.TabletHost, probe.TabletPort)
+		resp, err := throttler.httpClient.Get(tabletCheckSelfURL)
```

Comment:
Should we have a timeout? Maybe interval * 0.5 or something?

Reply:
The timeout is here:

https://github.com/vitessio/vitess/pull/9334/files#diff-aad1c740b216e391106ae8f24a0b47c524a4ee8c7c9d92e1936325712e26c1b2R191